### PR TITLE
Fix on typing event

### PR DESCRIPTION
**Pull request by xTEddie**
- Addressing #135 
- `onTypingEvent()` is not filtering user's own typing events

### CODE REVIEW

File: src/core/messaging/ACSClient.ts [hunk 1]
```diff
@@ -280,6 +280,13 @@ export class ACSConversation {
 
         try {
             const listener = (event: TypingIndicatorReceivedEvent) => {
+                const {sender, recipient} = event;
+
+                // Ignore participant's own typing events
+                if ((sender as any).communicationUserId === (recipient as any).communicationUserId) { // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (elopezanaya) [inline src/core/messaging/ACSClient.ts:286]: do we assume that those objects are always present including communicationUserId ?

File: playwright/integrations/unauthenticated-chat.spec.ts [hunk 2]
```diff
@@ -366,8 +366,8 @@ test.describe('UnauthenticatedChat @UnauthenticatedChat', () => {
         const {acsEndpoint, chatId} = runtimeContext;
         const getMessagesRequestPartialUrl = `${acsEndpoint}chat/threads/${encodeURIComponent(chatId)}/messages?api-version=`;
         const getMessagesResponseDataJson = await getMessagesResponse.json();
-        const sentMessages = getMessagesResponseDataJson.value.filter((message) => message.type === 'text');
-        const sentMessageContent = sentMessages[0].content.message;
+        const sentMessage = getMessagesResponseDataJson.value.filter((message) => message.type === 'text' && message.content.message === content);
+        const sentMessageContent = sentMessage[0].content.message;
```
Comment (elopezanaya) [inline playwright/integrations/unauthenticated-chat.spec.ts:370]: should we check for null?

**Review by elopezanaya — CHANGES_REQUESTED** [review]

File: src/core/messaging/ACSClient.ts (hunk 1 above)
Comment (akagar-microsoft) [inline src/core/messaging/ACSClient.ts:286]: +1 for null check. Also, can we add telemetry if these information is not present and is required for the operation.

File: playwright/integrations/unauthenticated-chat.spec.ts (hunk 2 above)
Comment (xTEddie) [reply playwright/integrations/unauthenticated-chat.spec.ts:370]: We shouldn't. It's an integration test to validate `ChatSDK.getMessages()`. In this test, we're sending a message first, then should expect the message that was sent to be part of the list of messages.

File: src/core/messaging/ACSClient.ts (hunk 1 above)
Comment (xTEddie) [reply src/core/messaging/ACSClient.ts:286]: It's a ACS typing event and should always contain a sender & receipient with its id.

Comment (xTEddie) [reply src/core/messaging/ACSClient.ts:286]: Yes since it's an ACS typing event and should always contain a sender & recipient with its id.

**Review by elopezanaya — APPROVED** [review]

**Pull request merged**